daphne/src/messages/mod.rs

```diff
@@ -125,42 +125,65 @@ pub type Time = u64;
 #[serde(rename_all = "snake_case")]
 #[cfg_attr(any(test, feature = "test-utils"), derive(deepsize::DeepSizeOf))]
 pub enum Extension {
-    Taskprov { payload: Vec<u8> }, // Not a TaskConfig to make computing the expected task id more efficient
-    Unhandled { typ: u16, payload: Vec<u8> },
+    Taskprov {
+        // Not a TaskConfig to make computing the expected task id more efficient
+        draft02_payload: Option<Vec<u8>>,
```

Comment:
In draft02, the taskprov extension includes a non-empty payload. In draft09, the taskprov extension includes an empty payload. Does that clarify?

Reply:
Not really, my thinking from the comment is that when that field is set, the vec contains a task id, so why not use the `TaskId` type instead of a Vec?

Reply:
I'll clarify the comment.

Reply:
Done.